sidestep_engine/core/trainer_helpers.py

```
@@ -268,6 +274,52 @@ def restore_rng_state(rng_state: Dict[str, Any], current_device: Any = None) ->
+            dev = torch.device(current_device)
+            current_idx = dev.index if dev.index is not None else 0
+        if saved_idx == current_idx:
+            try:
+                xpu_t = rng_state["xpu_rng"]
+                if isinstance(xpu_t, torch.Tensor):
+                    xpu_t = xpu_t.cpu()
+                    if xpu_t.dtype != torch.uint8:
+                        xpu_t = xpu_t.to(torch.uint8)
+                torch.xpu.set_rng_state(xpu_t, current_idx)
+                restored.append("xpu_rng")
+            except Exception as exc:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)